src/benchmarklib/benchmark_runner.cpp

```diff
@@ -155,7 +162,7 @@ void BenchmarkRunner::_benchmark_permuted_query_set() {
           }
         };
 
-        auto query_tasks = _schedule_or_execute_query(query_id, on_query_done);
+        auto query_tasks = _schedule_or_execute_query(query_id, false, on_query_done);
```

Comment:
You know what I think about literal boolean parameters and that I do not know what `_schedule_or_execute_query(query_id, false, on_query_done);` does. 

Something in the spirit of `_schedule_or_execute_query(query_id, IsPreparationQuery::Yes, on_query_done);` would be better, but then I'd ask myself why a function called "_schedule_or_execute_query" should care about whether the query its supposed to schedule_or_execute is a PREPARE query or not.

Reply:
I was planning to write a review comment about this, but forgot.

Originally I thought that splitting `_schedule_or_execute_query` into one method for preparation and another for execution would duplicate a lot of code, but now that you made me look at it again, I realize that there is no point in scheduling the preparation queries. I'll simply add a method `_prepare_query`.

Reply:
We wouldn't need the `prepare` parameter in `_schedule_or_execute_query` and `_schedule_query` anyway, right? It is only called with `false` everywhere.